plugin/trino-hive/src/main/java/io/trino/plugin/hive/projection/PartitionProjection.java

```diff
@@ -52,6 +53,44 @@ public PartitionProjection(Optional<String> storageLocationTemplate, Map<String,
         this.columnProjections = ImmutableMap.copyOf(requireNonNull(columnProjections, "columnProjections is null"));
     }
 
+    // TODO: support writing partition projection
+    public boolean allowWrite()
+    {
+        if (storageLocationTemplate.isPresent()) {
```

Comment:
I think this whole check for whether we can support writing boils down to `storageTemplateLocation.isEmpty()`. In the event that the custom template is provided but _happens_ to match the hive default template construction, the values would still need to be escaped in order to be created with the correct path.

Reply:
You are right, the logic for the template relates part has changed after this code has created.
Let's reject all the writes in case the `storageTemplateLocation` exists.